src/main/java/com/google/sps/servlets/ProfileServlet.java

```diff
@@ -106,7 +111,65 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+        String userId = userService.getCurrentUser().getUserId();
+        
+        // Don't update profile if cancel button is clicked
+        if (request.getParameter("submit").equals("Cancel")) {
+            response.sendRedirect("/profile.html?userID=" + userId);
+            return;
+        }
```

Comment:
Is it necessary to submit the post request when clicking cancel? Wouldn't the form just not be submitted?

Reply:
Since the cancel button is part of the form I think it submits no matter what. I tried making it a regular button that just hides the form when clicked, but that submitted the form anyway. This is the easiest thing I could find from the Google searches I did.